numpy/core/tests/test_multiarray.py

```diff
@@ -2041,6 +2041,24 @@ def __numpy_ufunc__(self, ufunc, method, pos, inputs, **kwargs):
         assert_raises(TypeError, np.dot, b, c)
         assert_raises(TypeError, c.dot, b)
 
+    def test_dot_type_mismatch(self):
+        c = 1.
+        A = np.array((1,1), dtype='i,i')
+
+        try:
```

Comment:
None, I wasn't sure the right way to do this in `numpy`'s test suite and took `test_mul` as an example.